src/plots/cartesian/set_convert.js

```diff
@@ -405,10 +405,15 @@ module.exports = function setConvert(ax, fullLayout) {
         if(axLetter in trace) {
             arrayIn = trace[axLetter];
             len = trace._length || arrayIn.length;
-            arrayOut = new Array(len);
 
-            for(i = 0; i < len; i++) {
-                arrayOut[i] = ax.d2c(arrayIn[i], 0, cal);
+            if(ax.type === 'linear' && Lib.isTypedArray(arrayIn) && arrayIn.subarray) {
+                arrayOut = arrayIn.subarray(0, len);
```

Comment:
More info: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/subarray

Bypassing `isNumeric(v) ? Number(v) : BADNAM` coercion saves about ~50ms at 1e4 points for SVG scatter. :racehorse:

Reply:
Typed arrays can't hold `undefined`, only `NaN`, right? So do we need to switch to `NaN` as `BADNUM`? Though this has the follow-on problem that `NaN === NaN` doesn't work, so `v === BADNUM` would have to be replaced by `IsNaN(v)` when we already know it's a number and I guess `IsNaN(v) && typeof v === 'number'` if we don't... That's pretty annoying. But if we DON'T, won't we have errors dealing with missing data in typed arrays?

Reply:
As far as I know, typed arrays "call" `Number(v)` on each item before "inserting" them into the typed array:

![image](https://user-images.githubusercontent.com/6675409/36794255-dd76728e-1c6d-11e8-8715-c6fd9d96535d.png)

so yeah (I think) that means `NaN`s are the only non-number values accepted in typed arrays.

Reply:
... so yeah `connectgaps: false` is probably broken for `scatter` with typed arrays. Let me check.

Reply:
Just for completeness while you're looking into edge cases, `(+/-)Infinity` are also allowed in float typed arrays. Aside from users directly providing `Infinity`, I presume this comes up in gl code when we drop doubles to singles, if the input is too large. I believe `(+/-)Infinity` and `NaN` are the only non-numerics that have representations in standard floating point encodings.

Reply:
To me surprise, `connectgaps: false` is working fine with typed arrays:

![image](https://user-images.githubusercontent.com/6675409/36803496-b2909e4c-1c85-11e8-82b4-a13f39a660ea.png)

----------------

Here are all the places we check equality with `BADNUM`:


![image](https://user-images.githubusercontent.com/6675409/36803533-ce34635e-1c85-11e8-9e46-f18c00d45d16.png)

by the looks of it, these are all places that check `gd.calcdata` values (i.e. not in a typed array). Luckily `isNumeric` is used elsewhere (which works for fine detecting `NaN`s in typed arrays`).

I'll test out a few more case this afternoon.

Reply:
All right, I couldn't find an example of `BADNUM` breaking things for typed array. 

Replacing `BADNUM`s with `NaN`s (and `isNaN`) might help us be more  _typed-array_ friendly down the road, but as long as we're using arrays of object of our `calcdata` for most things, `BADNUM` should work just fine. 

Note that these two [lines](https://github.com/plotly/plotly.js/blob/d4cb0c43570c7619d68af75da93475822a17871a/src/traces/scattergl/index.js#L65-L66) in scattergl calc aren't necessary (as `Number(BADNUM) // -> NaN`), but it makes things clear that `BADNUM` and typed array aren't the best of friends, so I'll keep it there.